addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeGatewayHandler.java

```diff
@@ -0,0 +1,303 @@
+        if (config.hostname == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Gateway hostname must be set");
+        } else if (config.port == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Gateway port must be set");
+        } else if (config.gatewayId == null || SinopeConfig.convert(config.gatewayId) == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Gateway Id must be set");
+        } else if (config.apiKey == null || SinopeConfig.convert(config.apiKey) == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Api Key must be set");
+        } else {
+            refreshInterval = config.refresh;
+            schedulePoll();
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+        if (clientSocket != null) {
+            try {
+                clientSocket.close();
+            } catch (IOException e) {
+                logger.warn("Unexpected error when closing connection to gateway", e);
+            }
+        }
+    }
+
+    synchronized void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
+        pollFuture = scheduler.scheduleAtFixedRate(pollingRunnable, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private synchronized void poll() {
+        if (thermostatHandlers.size() > 0) {
+            logger.debug("Polling for state");
+            try {
+                if (connectToBridge()) {
+                    logger.debug("Connected to bridge");
+                    for (SinopeThermostatHandler sinopeThermostatHandler : thermostatHandlers) {
+                        sinopeThermostatHandler.update();
+                    }
+                }
+            } catch (IOException e) {
+                logger.error("Could not connect to gateway", e);
+                setCommunicationError(true);
+            } finally {
+
+            }
+        } else {
+            logger.debug("nothing to poll");
+        }
+    }
+
+    boolean connectToBridge() throws UnknownHostException, IOException {
+        SinopeConfig config = getConfigAs(SinopeConfig.class);
+        if (this.clientSocket == null || !this.clientSocket.isConnected() || this.clientSocket.isClosed()) {
+            this.clientSocket = new Socket(config.hostname, config.port);
+            SinopeApiLoginRequest loginRequest = new SinopeApiLoginRequest(SinopeConfig.convert(config.gatewayId),
```

Comment:
The condition (socket == null or not Connectected or closed then new Socket) is checking first if the socket is still open and it shouldn't recreate a connection. Are you seeing something else? Gateway is single threaded and accept one connection..  I can see openhab keeping the connection because I need to close my production box to connect with with dev box.

Reply:
Sorry, never mind I was wrong